src/python/pants/console/stty_utils.py

```diff
@@ -5,33 +5,43 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import logging
+import sys
+import termios
+import tty
 from contextlib import contextmanager
 
-from pants.util.process_handler import subprocess
 
-
-@contextmanager
-def preserve_stty_settings():
-  """Run potentially stty-modifying operations, e.g., REPL execution, in this contextmanager."""
-  stty_settings = STTYSettings()
-  stty_settings.save_stty_options()
-  yield
-  stty_settings.restore_ssty_options()
+logger = logging.getLogger(__name__)
 
 
 class STTYSettings(object):
   """Saves/restores stty settings, e.g., during REPL execution."""
 
-  def __init__(self):
-    self._stty_options = None
-
-  def save_stty_options(self):
-    self._stty_options = self._run_cmd('stty -g 2>/dev/null')
+  @classmethod
+  @contextmanager
+  def preserved(cls):
+    """Run potentially stty-modifying operations, e.g., REPL execution, in this contextmanager."""
+    inst = cls()
```

Comment:
Seems a little odd for this to be a classmethod which instantiates itself, rather than the caller calling `with STTYSettings().preserved():`, but not a big deal

Reply:
was thinking in a similar vein to `Class.open()` or `Class.create()` to avoid lifecycle misuse.